### PR TITLE
fix: tensor dimension swap instead of view change

### PR DESCRIPTION
**Pull request by ab93**

### CODE REVIEW

File: numalogic/tools/data.py [hunk 1]
```diff
@@ -179,6 +182,30 @@ def __getitem__(self, idx: Union[int, slice]) -> npt.NDArray[float]:
         return self._data[idx : idx + self._seq_len]
 
 
+class StreamingDataLoader(DataLoader):
+    """
+    A DataLoader for convinience that uses StreamingDataset for handling time series data.
```
Comment (qhuai) [inline numalogic/tools/data.py:187]: typo: convenience.

File: numalogic/models/autoencoder/variants/vanilla.py [hunk 2]
```diff
@@ -182,25 +182,25 @@ def __init__(
...
     @staticmethod
     def init_weights(m: nn.Module) -> None:
-        r"""Initiate parameters in the transformer model."""
+        """Initialize the parameters in the model."""
         if type(m) == nn.Linear:
             nn.init.xavier_normal_(m.weight)
 
     def forward(self, batch: Tensor) -> tuple[Tensor, Tensor]:
-        batch = batch.view(-1, self.n_features, self.seq_len)
+        batch = torch.swapdims(batch, 1, 2)
```
Comment (qhuai) [inline numalogic/models/autoencoder/variants/vanilla.py:190]: Earlier, when `batch.view` was applied, each feature vector sent to each channel carries datapoints interleaved from different metrics (such as error%, cpu, memory). Now, with `torch.swapdims`, are we merely sending datapoints of different metrics to different channels?

If the answer is 'yes', we would be detecting pattern mismatches for each metrics independent of the others.

Comment (ab93) [reply numalogic/models/autoencoder/variants/vanilla.py:190]: I think you are misunderstanding the requirement here. We are moving/swapping the dims because the convolutional operation requires the tensor to be aligned in a particular dimensional consistency. https://pytorch.org/docs/stable/generated/torch.nn.Conv1d.html
Regarding the dense layers, the contraction on the dimensions is happening on the sequence length dim, not on the feature dim.
